### PR TITLE
Add python-json-logger and change default log format to JSON

### PR DESCRIPTION
**Pull request by isugimpy**

### CODE REVIEW

File: baseplate/lib/log_formatter.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from pythonjsonlogger import jsonlogger
+
+
+class CustomJsonFormatter(jsonlogger.JsonFormatter):
+    def process_log_record(self, log_record: dict) -> dict:
+        log_record["level"] = log_record.pop("levelname", None)
```
[the remark this answers is not in the document]
Comment (isugimpy) [reply baseplate/lib/log_formatter.py:6]: This is specifically to handle LogDNA compatibility.  They expect the name of the log level field to be `level`, but Python presents it as `levelname`, so this is just doing that mutation.

File: setup.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
     install_requires=[
         "posix_ipc>=1.0.0",
         "pyjwt>=1.6.0",
+        "python-json-logger==0.1.11",
```
[the remark this answers is not in the document]
Comment (isugimpy) [reply setup.py:19]: The reason I went with the exact pin is that the next version release according to the developer is `2.0`, and I didn't want us to have any surprises.  We could do `~=0.1` if you'd prefer that, but there's no reason to expect that there will be a `0.1.12`.

File: baseplate/lib/log_formatter.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+        return jsonlogger.JsonFormatter.process_log_record(self, log_record)
```
[the remark this answers is not in the document]
Comment (isugimpy) [reply baseplate/lib/log_formatter.py:7]: Does the above actually work as expected?  I could have sworn I tried that and it somehow didn't, but I'm 100% open to switching it to your version.

File: baseplate/lib/log_formatter.py (hunk 1 above)
Comment (isugimpy) [inline baseplate/lib/log_formatter.py:6]:
```suggestion
        log_record["level"] = log_record.pop("levelname", None)
        if "traceID" not in log_record:
            log_record["traceID"] = log_record.get("threadName", None)
```

**Comment by isugimpy** [comment]
:haircut_man: 

I also threw in the traceID thing.

File: baseplate/lib/log_formatter.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from pythonjsonlogger import jsonlogger
+
+
+class CustomJsonFormatter(jsonlogger.JsonFormatter):
+    def process_log_record(self, log_record: dict) -> dict:
+        log_record["level"] = log_record.pop("levelname", None)
+        if "traceID" not in log_record:
```
Comment (isugimpy) [inline baseplate/lib/log_formatter.py:7]:
```suggestion
        if "traceID" not in log_record and "threadName" in log_record:
```

File: baseplate/lib/log_formatter.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from pythonjsonlogger import jsonlogger
+
+
+class CustomJsonFormatter(jsonlogger.JsonFormatter):
+    def process_log_record(self, log_record: dict) -> dict:
+        log_record["level"] = log_record.pop("levelname", None)
+        if "traceID" not in log_record and "threadName" in log_record:
+            log_record["traceID"] = log_record.get("threadName", None)
```
Comment (spladug) [inline baseplate/lib/log_formatter.py:8]: you need to make sure it's an integer to call it a traceID. there can be greenlets with other thread names that would make traceID inaccurate.

File: baseplate/lib/log_formatter.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (spladug) [reply baseplate/lib/log_formatter.py:6]: got it, thanks

File: setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (spladug) [reply setup.py:19]: ~=0.1 sgtm

Comment (isugimpy) [inline setup.py:19]:
```suggestion
        "python-json-logger~=0.1",
```

File: baseplate/lib/log_formatter.py (hunk 5 above)
Comment (isugimpy) [inline baseplate/lib/log_formatter.py:8]:
```suggestion
        try:
            log_record["traceID"] = int(log_record.get("threadName", None))
        except ValueError:
            pass
```

File: baseplate/lib/log_formatter.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
+from pythonjsonlogger import jsonlogger
+
+
+class CustomJsonFormatter(jsonlogger.JsonFormatter):
+    def process_log_record(self, log_record: dict) -> dict:
+        log_record["level"] = log_record.pop("levelname", None)
+        if "traceID" not in log_record and "threadName" in log_record:
```
Comment (isugimpy) [inline baseplate/lib/log_formatter.py:7]:
```suggestion
```

File: requirements-test.txt [hunk 7]
```diff
@@ -38,6 +38,7 @@ pluggy==0.12.0
 posix-ipc==1.0.4
 py==1.7.0
 pycparser==2.19
+python-json-logger==0.1.11
```
Comment (isugimpy) [inline requirements-test.txt:41]:
```suggestion
python-json-logger~=0.1
```

File: baseplate/lib/log_formatter.py [hunk 8]
```diff
@@ -0,0 +1,11 @@
+from pythonjsonlogger import jsonlogger
+
+
+class CustomJsonFormatter(jsonlogger.JsonFormatter):
+    def process_log_record(self, log_record: dict) -> dict:
+        log_record["level"] = log_record.pop("levelname", None)
+        try:
+            log_record["traceID"] = int(log_record.get("threadName", None))
+        except ValueError:
```
Comment (isugimpy) [inline baseplate/lib/log_formatter.py:9]:
```suggestion
        except (TypeError, ValueError):
```

File: baseplate/lib/log_formatter.py [hunk 9]
```diff
@@ -0,0 +1,11 @@
+from pythonjsonlogger import jsonlogger
+
+
+class CustomJsonFormatter(jsonlogger.JsonFormatter):
+    def process_log_record(self, log_record: dict) -> dict:
+        log_record["level"] = log_record.pop("levelname", None)
+        try:
+            log_record["traceID"] = int(log_record.get("threadName", None))
+        except (TypeError, ValueError):
```
Comment (spladug) [inline baseplate/lib/log_formatter.py:9]:
```suggestion
            log_record["traceID"] = int(log_record["threadName"])
        except (KeyError, ValueError):
```

this feels a bit more clear to me and skips trying to feed a `None` to `int()`

**Pull request merged**